test/clj/rems/api/test_applications.clj

```diff
@@ -256,102 +292,36 @@
 (deftest dynamic-application-create-test
   (let [api-key "42"
         user-id "alice"
-        catid 9 ;; catalogue item with dynamic workflow in test-data
-        draft (create-application-draft-for-catalogue-item 9)]
-    (testing "get draft"
-      (is (< 6 (count (:items draft)))))
-    (let [response (-> (request :post (str "/api/applications/save"))
-                       (authenticate api-key user-id)
-                       (json-body {:command "save"
-                                   :catalogue-items [catid]
-                                   :items {1 "dynamic test"}})
-                       app
-                       read-body)
-          application-id (:id response)]
-      (testing "create application"
-        (is (some? application-id))
-        (let [saved (get-application user-id application-id)]
-          (is (= "workflow/dynamic" (get-in saved [:application :workflow :type])))
-          (is (= "rems.workflow.dynamic/draft" (get-in saved [:application :state])))
-          (is (= "dynamic test" (get-in saved [:items 0 :value])))))
-      (testing "getting application as other user is forbidden"
-        (is (response-is-forbidden?
-             (-> (request :get (str "/api/v2/applications/" application-id))
-                 (authenticate api-key "bob")
-                 app))))
-      (testing "saving fields as other user is forbidden"
-        (is (response-is-forbidden?
-             (-> (request :post "/api/applications/save")
-                 (authenticate api-key "bob")
-                 (json-body {:command "save"
-                             :application-id application-id
-                             :items {1 "xxxxx"}})
-                 app))))
-      (testing "can't submit with missing required fields"
```

Comment:
where did these validation tests go?

Reply:
It's covered by https://github.com/CSCfi/rems/blob/dead-code/src/clj/rems/workflow/dynamic.clj#L938-L960 and the test for maxlength was made into a unit test at https://github.com/CSCfi/rems/blob/dead-code/test/clj/rems/test_form_validation.clj#L49-L59